src/udev/udev-builtin-net_id.c

```diff
@@ -821,6 +823,19 @@ static int names_devicetree(sd_device *dev, const char *prefix, EventMode mode)
         ofnode_path--;
         assert(path_is_absolute(ofnode_path));
 
+        if (naming_scheme_has(NAMING_DEVICETREE_ETHERNET_PORTS)) {
+                /* Check if there are 'ethernet-ports' child nodes of ethernet */
+                r = sd_device_new_child(&eth_ports_dev, ofnode_dev, "ethernet-ports");
+                if (r < 0) {
+                        log_device_debug(ofnode_dev, "No 'ethernet-ports' child available for this device");
+                        /* 'ports' can also be used as node name. If 'ethernet-ports'
+                         * was not found, look for 'ports' instead */
+                        r = sd_device_new_child(&eth_ports_dev, ofnode_dev, "ports");
+                        if (r < 0)
+                                log_device_debug(ofnode_dev, "No 'ports' child available for this device");
+                }
```

Comment:
Not a big matter, but I think it is OK to merge two debugging logs. Please use `log_device_debug_errno()`, and mention the failure will be ignored, as this is optional.
```c
/* Check if there are 'ethernet-ports' child nodes of ethernet */
r = sd_device_new_child(&eth_ports_dev, ofnode_dev, "ethernet-ports");
if (r < 0)
        /* 'ports' can also be used as node name. If 'ethernet-ports'
         * was not found, look for 'ports' instead */
        r = sd_device_new_child(&eth_ports_dev, ofnode_dev, "ports");
if (r < 0)
        log_device_debug_errno(ofnode_dev, r, "Neither 'ethernet-ports' nor 'ports' child available, ignoring: %m");
```